frontend/src/providers/Dashboard/Dashboard.tsx

```diff
@@ -0,0 +1,118 @@
+import get from 'api/dashboard/get';
+import { PANEL_TYPES } from 'constants/queryBuilder';
+import { REACT_QUERY_KEY } from 'constants/reactQueryKeys';
+import ROUTES from 'constants/routes';
+import {
+	createContext,
+	PropsWithChildren,
+	useContext,
+	useMemo,
+	useState,
+} from 'react';
+import { Layout } from 'react-grid-layout';
+import { useQuery, UseQueryResult } from 'react-query';
+import { useSelector } from 'react-redux';
+import { useLocation, useRouteMatch } from 'react-router-dom';
+import { AppState } from 'store/reducers';
+import { Dashboard } from 'types/api/dashboard/getAll';
+import AppReducer from 'types/reducer/app';
+
+import { IDashboardContext } from './types';
+
+const DashboardContext = createContext<IDashboardContext>({
+	isDashboardSliderOpen: false,
+	handleToggleDashboardSlider: () => {},
+	dashboardResponse: {} as UseQueryResult<Dashboard, unknown>,
+	selectedDashboard: {} as Dashboard,
+	dashboardId: '',
+	layouts: [],
+	setLayouts: () => {},
+	setSelectedDashboard: () => {},
+});
+
+export function DashboardProvider({
+	children,
+}: PropsWithChildren): JSX.Element {
+	const [isDashboardSliderOpen, setIsDashboardSlider] = useState<boolean>(false);
+	const { pathname } = useLocation();
+	const isDashboardPage = useRouteMatch({
+		path: ROUTES.DASHBOARD,
+		exact: true,
+	});
+
+	const isDashboardWidgetPage = useRouteMatch({
+		path: ROUTES.DASHBOARD_WIDGET,
+		exact: true,
+	});
+
+	const [layouts, setLayouts] = useState<Layout[]>([]);
+
+	const { isLoggedIn } = useSelector<AppState, AppReducer>((state) => state.app);
+
+	const pathnameArray = pathname.split('/');
+	const dashboardId = pathnameArray[pathnameArray.length - 1];
```

Comment:
We should start using uuid for ID's , depending on array index will land us in errors if not handled properly.

Reply:
fixed